pkg/mcoproducer/mcoproducer.go

```diff
@@ -0,0 +1,146 @@
+package mcoproducer
+
+import (
+	"encoding/base64"
+	"fmt"
+
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/rh-ecosystem-edge/kernel-module-management/internal/utils"
+)
+
+const replaceKernelModuleScriptTemplate = `#!/bin/bash
+echo "before checking podman images"
+if podman images | grep $IMAGE | grep -q $TAG; then
+    echo "Image $IMAGE:$TAG found in the local registry, removing in-tree kernel module"
+    podman run --privileged --entrypoint modprobe $IMAGE:$TAG -rd /opt $KERNEL_MODULE
```

Comment:
```suggestion
    echo "Image ${IMAGE}:${TAG} found in the local registry, removing in-tree kernel module"
    podman run --privileged --entrypoint modprobe ${IMAGE}:${TAG} -rd /opt $KERNEL_MODULE
```

Just to be on the safe side.

Reply:
Agree regarding parenthesis, but i don't agree regarding entrypoint: we don't know how that container was defined, and in case it does contain COMMAND or ENTRYPOINT directive we need to explicitly override it